src/win/process.c

```diff
@@ -1250,7 +1252,69 @@ static int uv__kill(HANDLE process_handle, int signum) {
     return UV_EINVAL;
   }
 
+  /*
+   * Create a dump file for the targeted process, if the registry key
+   * `HKLM:Software\Microsoft\Windows\Windows Error Reporting\LocalDumps\DumpFolder`
+   * points to a valid folder.  The default value if the registry key does not
+   * exist is `%LOCALAPPDATA%\CrashDumps`, see [0] for more detail.
+   * [0]: https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps
+   */
+  if (signum == SIGQUIT) {
+    /* Get target process name. */
+    WCHAR basename[MAX_PATH];
+    GetModuleBaseNameW(process_handle, NULL, &basename[0], sizeof(basename));
+
+    /* Get PID of target process. */
+    DWORD pid = GetProcessId(process_handle);
+
+    /* Get LocalDumps directory path. */
+    HKEY registry_key;
+    WCHAR dump_folder[MAX_PATH];
+    DWORD ret = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                  L"SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps",
+                  0,
+                  KEY_QUERY_VALUE,
+                  &registry_key);
+    if (ret == ERROR_SUCCESS) {
+      DWORD dump_folder_len = sizeof(dump_folder);
+      DWORD key_type = 0;
+      ret = RegGetValueW(registry_key,
+                          NULL,
+                          L"DumpFolder",
+                          RRF_RT_ANY,
+                          &key_type,
+                          (PVOID) dump_folder,
+                          &dump_folder_len);
+      if (ret != ERROR_SUCCESS) {
+        /* Default value for `dump_folder` is `%LOCALAPPDATA%\CrashDumps`. */
+        WCHAR * localappdata;
+        SHGetKnownFolderPath(&FOLDERID_LocalAppData, 0, NULL, &localappdata);
+        _snwprintf(dump_folder, sizeof(dump_folder), L"%ls\\CrashDumps", localappdata);
+        CoTaskMemFree(localappdata);
+      }
+      RegCloseKey(registry_key);
+
+    /* Construct dump filename. */
+    WCHAR dump_name[MAX_PATH];
+    swprintf_s(dump_name, sizeof(dump_name), L"%ls\\%ls.%d.dmp", dump_folder, basename, pid);
+
+    HANDLE hDumpFile = CreateFileW(dump_name, GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (hDumpFile != INVALID_HANDLE_VALUE) {
+      /* Tell wine to dump ELF modules as well */
+      DWORD sym_options = SymGetOptions();
+      SymSetOptions(sym_options | 0x40000000);
+      MiniDumpWriteDump(
+        process_handle, pid,
+        hDumpFile, MiniDumpWithFullMemory | MiniDumpIgnoreInaccessibleMemory | MiniDumpWithAvxXStateContext,
+        NULL, NULL, NULL
+      );
```

Comment:
```suggestion
      if (MiniDumpWriteDump(
        process_handle, pid,
        hDumpFile, MiniDumpWithFullMemory | MiniDumpIgnoreInaccessibleMemory | MiniDumpWithAvxXStateContext,
        NULL, NULL, NULL
      )) {
        FILE_DISPOSITION_INFO DeleteOnClose = { FALSE };
        SetFileInformationByHandle(hDumpFile, FileDispositionInfo, &DeleteOnClose, sizeof(DeleteOnClose);
      }
```
(and also open the file with `FILE_FLAG_DELETE_ON_CLOSE`)